include/xtensor/xdisk_array.hpp

```diff
@@ -0,0 +1,168 @@
+#ifndef XTENSOR_DISK_ARRAY_HPP
+#define XTENSOR_DISK_ARRAY_HPP
+
+#include "xarray.hpp"
+#include "xcsv.hpp"
+
+namespace xt
+{
+    template <class T>
+    class xdisk_reference
+    {
+    public:
+
+        xdisk_reference(T& value)
+        {
+            m_pvalue = &value;
+        }
+
+        operator const T() const
+        {
+            return *m_pvalue;
+        }
+
+        operator T()
+        {
+            return *m_pvalue;
+        }
+
+        T operator=(const T value)
+        {
+            *m_pvalue = value;
+            dump_csv(*m_out_file, *m_array);
+            return *m_pvalue;
+        }
+
+        friend std::ostream& operator<<(std::ostream& os, const xdisk_reference<T>& obj)
+        {
+            return os << *obj.m_pvalue;
+        }
+
+        static std::ifstream* m_in_file;
+        static std::ofstream* m_out_file;
+        const static xarray<T>* m_array;
```

Comment:
I am really not sure about having these static members. The `xdisk_reference` class is here to catch assignments to elements of `m_array` through its definition of the `operator=()`, by making `xdisk_reference` the type of `xdisk_array`'s `reference`. But it needs to access `m_array` and the output file in order to dump to disk, and they cannot be passed through e.g. the constructor since (as I understand it) the copy constructor is used.

Reply:
You can explicitly invoke the constructor of `xdisk_reference` and pass it whatever you want. For instance:

```cpp
template <class It>
inline reference element(It first, It last)
{
    return reference(m_array->element(first, last), &m_array, &m_in_file, &m_out_file);
}
```

Reply:
Awesome!